kernel/src/hil/eic.rs

```diff
@@ -0,0 +1,55 @@
+//! Interface for external interrupt controller.
```

Comment:
Having some more description here that provides future implementors some intuition on _when_ to use this HIL would be helpful.

Reply:
This is a question I need to explore a bit more: to figure out why application developers might want to use EIC. Right now the reason seems like an external interrupt in async mode can wake up the system from the deep sleep mode, whereas a GPIO pin interrupt cannot. If that's the case, then the question goes to "why not just set it always in async mode?". I'll add these as comments.

Reply:
Well that is likely specific to the SAM4L (or maybe some subset of microcontrollers), but the ability to configure interrupt sources at this low level might be more generic. In particular, I think it would be helpful to guide the developer that is trying to add Tock support to a new MCU, and who needs to map hardware peripherals to HIL interfaces. I'm less worried about users higher up the stack that are trying to develop for, say, imix, and need to know when to use EIC vs. GPIO.